src/runtime/pkg/containerd-shim-v2/service.go

```diff
@@ -925,6 +925,11 @@ func (s *service) Shutdown(ctx context.Context, r *taskAPI.ShutdownRequest) (_ *
 	shimLog.WithField("container", r.ID).Debug("Shutdown() start")
 	defer shimLog.WithField("container", r.ID).Debug("Shutdown() end")
 	span, _ := katatrace.Trace(s.rootCtx, shimLog, "Shutdown", shimTracingTags)
+	defer func() {
+		if span != nil {
```

Comment:
I agree that we don't need to check for nil here

Reply:
@liubin @amshinde  @cmaf thank you for your comments, I check the `span != nil`, because the man process exec the `katatrace.StopTracing(s.rootCtx)`, it would destroy the `span` Object.

Reply:
@YchauWang If this is only for the early return of line 943, I think you can only add `span.End()` before 943.

Reply:
Thanks @liubin, and one question as follow comment of cmaf, can the `shutdown` command replay many times?

Reply:
@lifupan can you help to answer this?

Reply:
@YchauWang  @liubin No, the containerd would only call the shutdown only one time, and there's no replay.

Reply:
Thanks @lifupan @liubin, I think we can use @cmaf 's suggestion of that comment.